syntax = "proto3";

option go_package="scripts";

package com.ibm.cloudland.scripts;

service RemoteExec {
  rpc Execute (ExecuteRequest) returns (ExecuteReply) {}
  rpc Transmit (stream FileChunk) returns (TransmitAck) {}
}

message ExecuteRequest {
  int32 id = 1;
  string control = 2;
  string command = 3;
  int32 extra = 4;
}

message ExecuteReply {
  string status = 1;
}

message FileChunk {
  int32 id = 1;
  string control = 2;
  string filepath = 3;
  int32 filesize = 4;
  int32 checksum = 5;
  int64 fileseek = 6;
  bytes content = 7;
  int32 extra = 8;
}

message TransmitAck {
  string status = 1;
}
